src/PortingAssistant.Compatibility.Common/Utils/RegionalDatastoreService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Amazon;
+using Amazon.S3;
+using PortingAssistant.Compatibility.Common.Interface;
+using Amazon.S3.Model;
+using Microsoft.Extensions.Logging;
+
+namespace PortingAssistant.Compatibility.Common.Utils
+{
+    public class RegionalDatastoreService : IRegionalDatastoreService
+    {
+        private readonly IHttpService _httpService;
+        private readonly AmazonS3Client _s3Client;
+        private readonly bool _isLambdaEnvSetup;
+        private readonly string _regionaS3BucketName;
+        private readonly ILogger<RegionalDatastoreService> _logger;
+
+        public RegionalDatastoreService(
+            IHttpService httpService,
+            ILogger<RegionalDatastoreService> logger
+            )
+        {
+            _httpService = httpService;
+            _logger = logger;
+            string region = Environment.GetEnvironmentVariable("AWS_REGION");
+
+            if (!string.IsNullOrEmpty(region)) 
+            {
+                _isLambdaEnvSetup = true;
+                _regionaS3BucketName = $"portingassistant-datastore-{region}"; 
+                _s3Client = new AmazonS3Client(RegionEndpoint.GetBySystemName(region));
+            }
+        }
+
+        public async Task<Stream> DownloadGitHubFileAsync(string fileToDownload)
+        {
+            return await _httpService.DownloadGitHubFileAsync(fileToDownload);
+        }
+
+        public async Task<Stream?> DownloadRegionalS3FileAsync(string fileToDownload, bool isRegionalCall = false)
+        {
+            try 
+            {
+                _logger.LogInformation($"Downloading {fileToDownload} from regional S3 " + _regionaS3BucketName);
+                if (isRegionalCall && _isLambdaEnvSetup)
+                {
+                    GetObjectRequest request = new GetObjectRequest
+                    {
+                        BucketName = _regionaS3BucketName,
+                        Key = fileToDownload
+                    };
+                    using (GetObjectResponse response = await _s3Client.GetObjectAsync(request))
```

Comment:
if GetObjectResponse fails using S3Client,  call _httpService.DownloadS3FileAsync. Check the  response status value.